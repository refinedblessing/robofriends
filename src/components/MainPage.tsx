import * as React from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import CounterButton from './CounterButton';

const { useEffect } = React;

const filteredRobots = (robots, searchTerm) => robots.filter(({ name }) => name.toLowerCase().includes(searchTerm.toLowerCase()))
const MainPage = ({ onSearchChange, searchTerm, robots, onRequestRobots, isPending }) => {
  useEffect(() => {
    onRequestRobots();
  }, [onRequestRobots])

  return (
    <div className='tc'>
      <CounterButton color={'blue'}/>
      <h1>RoboFriends</h1>
      <SearchBox searchTerm={searchTerm} onChange={onSearchChange}/>
      {filteredRobots.length ? 
        <Scroll>
          <CardList robots={filteredRobots(robots, searchTerm)}/>
        </Scroll> : <h2>NO Robots</h2>}
    </div>
  )
}

export default MainPage;
