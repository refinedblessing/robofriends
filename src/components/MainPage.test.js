import { shallow } from 'enzyme'
import MainPage from './MainPage';
import React from 'react'

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    onSearchChange: jest.fn(),
    robots: [],
    searchTerm: '',
    isPending: false,
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
})

it('should expect Mainpage to render', () => {
  expect(wrapper).toMatchSnapshot()
})

