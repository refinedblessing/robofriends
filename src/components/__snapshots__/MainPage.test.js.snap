// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should expect Mainpage to render 1`] = `
<div
  className="tc"
>
  <CounterButton />
  <h1>
    RoboFriends
  </h1>
  <SearchBox
    onChange={[MockFunction]}
    searchTerm=""
  />
  <Scroll>
    <CardList
      robots={Array []}
    />
  </Scroll>
</div>
`;
